src/pipes.cpp

```diff
@@ -0,0 +1,66 @@
+#include "pipes.h"
+
+const int Pipes::NUM_PIPES = 4;
+const float Pipes::PIPE_SCALE = 0.4;
+const int Pipes::PIPE_OFFSET_RANGE = 300;
+const float Pipes::SPEED_PER_SECOND = 500;
+
+Pipes::Pipes(int screenWidth, int screenHeight, int frameRate)
+{
+    bool res = pipeTexture.loadFromFile("./lib/pipe.png");
+    if(!res) throw("Cannot load pipe\n");
+
+    pipeSprites = new sf::Sprite[NUM_PIPES];
+    if (pipeSprites == nullptr) throw("Pipe array could not be created\n");
+
+    for(int i = 0 ; i < NUM_PIPES ; i++)
+    {
+        this->pipeSprites[i].setTexture(pipeTexture, true);
+        this->pipeSprites[i].setScale(PIPE_SCALE, PIPE_SCALE);
+
+        if(i % 2 == 1) this->pipeSprites[i].setRotation(180);
+
+        sf::Vector2f dimensions = this->pipeSprites[i].getLocalBounds().getSize();
+        this->pipeSprites[i].setOrigin(dimensions.x / 2, dimensions.y / 2);
+
+        sf::Vector2f position;
+        bool isFirstPipe = int(i / 2) == 0;
+        position.x = screenWidth * (isFirstPipe ? 3 : 1) / 4 + int(isFirstPipe) * dimensions.x / 2;
+        position.y = screenHeight * (i % 2 == 0 ? 6 : 0) / 6;
+        
+        this->pipeSprites[i].setPosition(position);
+    }
+
+    this->frameRate = frameRate;
+    this->speed = SPEED_PER_SECOND / frameRate;
```

Comment:
SPEED_PER_SECOND / framerate is okay.. but check out the "Fix your timestep" blog in another page to learn how people tend to do speeds in video games. Typically we use a "variable" timestep so every frame has a different deltatime and we just store a speed value in units / second, multiply that by the delta time for the frame to get units / frame and then use that value instead of assuming a fixed rate.